classes/form/CustomerPersister.php

```diff
@@ -188,6 +188,19 @@ private function create(Customer $customer, $clearTextPassword)
             return false;
         }
 
+        /**
+         * Database optimization
+         * We check if the email has been assigned to guest account
+         */
+        $check = Customer::customerExists($customer->email, true, false);
+        if ($check) {
+            $customer = new Customer($check);
+            $this->context->updateCustomer($customer);
+            $this->context->cart->update();
+
+            return true;
```

Comment:
I guess we could only replace the variable `$customer` and reuse the end of this method `create(...)`.

Reply:
I mean in my opinion this code would be enough here:

```php
        if ($customer_id) {
            $customer = new Customer($customer_id);
        }